### PR TITLE
Migrated db

### PR DESCRIPTION
**Pull request by jonasfj**
cc @isoos for review? if you have time...

Disclaimer: I think the first commit migrating `datastore` is solid, I did that by hand... the second one got a little help from the migration tool. There could be bugs, and probably could be things better expressed differently.

### CODE REVIEW

File: lib/src/db/db.dart [hunk 1]
```diff
@@ -237,15 +240,24 @@ class Query<T extends Model> {
...
     if (_partition != null) {
-      partition = ds.Partition(_partition.namespace);
+      partition = ds.Partition(_partition!.namespace);
     }
 
     return StreamFromPages<ds.Entity>((int pageSize) {
-      return _db.datastore
-          .query(query, transaction: _transaction, partition: partition);
-    }).stream.map<T>(_db.modelDB.fromDatastoreEntity);
+      if (_transaction != null) {
```
Comment (isoos) [inline lib/src/db/db.dart:249]: Nit: why was this needed? I'd thought that you could have passed `transaction: _transaction, partition: partition` as before.

File: lib/src/db/db.dart [hunk 2]
```diff
@@ -265,12 +277,11 @@ class Query<T extends Model> {
...
-  Partition _defaultPartition;
+  final Partition _defaultPartition;
 
-  DatastoreDB(this.datastore, {ModelDB modelDB, Partition defaultPartition})
-      : _modelDB = modelDB ?? ModelDBImpl() {
-    _defaultPartition = defaultPartition ?? Partition(null);
-  }
+  DatastoreDB(this.datastore, {ModelDB? modelDB, Partition? defaultPartition})
+      : _modelDB = modelDB ?? ModelDBImpl(),
+        _defaultPartition = defaultPartition ?? Partition(null);
```
Comment (isoos) [inline lib/src/db/db.dart:284]: Nit: `Partition.DEFAULT`?

File: lib/src/db/db.dart [hunk 3]
```diff
@@ -402,14 +414,21 @@ Future _commitHelper(DatastoreDB db,
   if (deletes != null) {
     entityDeletes = deletes.map(db.modelDB.toDatastoreKey).toList();
   }
+  Future<ds.CommitResult> r;
+  if (datastoreTransaction != null) {
+    r = db.datastore.commit(
+        inserts: entityInserts ?? [],
+        autoIdInserts: entityAutoIdInserts ?? [],
+        deletes: entityDeletes ?? [],
+        transaction: datastoreTransaction);
```
Comment (isoos) [inline lib/src/db/db.dart:423]: Nit: same here: why can't we pass `transaction: datastoreTransaction`?

**Review by isoos — APPROVED** [review]
A few nits, but lgtm otherwise.

File: lib/src/db/db.dart (hunk 1 above)
Comment (jonasfj) [reply lib/src/db/db.dart:249]: We can, if we allow `transaction` to be nullable... but is that a better choice? maybe it's fine...

File: lib/src/db/db.dart (hunk 2 above)
Comment (jonasfj) [reply lib/src/db/db.dart:284]: I think that's `datastore.Partition` and not `db.Partition`, yes, I'm also confused..

File: lib/src/db/db.dart (hunk 3 above)
Comment (jonasfj) [reply lib/src/db/db.dart:423]: because if we allow `transaction: datastoreTransaction` then `transaction` have to be nullable..

And we might prefer that users don't pass a `datastoreTransaction` that they forgot to initialize.
